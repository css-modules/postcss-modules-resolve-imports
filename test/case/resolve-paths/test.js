'use strict';

const setup = require('../../setup');
const resolveImports = require('../../../index');

test('resolve-paths', () => {
  const {resulting} = setup(
    'local-by-default',
    'extract-imports',
    'scope',
    'self'
  )(__dirname);

  expect(resulting).toMatchSnapshot();
});

test('resolve-paths custom root', () => {
  const {resulting} = setup(
    'local-by-default',
    'extract-imports',
    'scope',
    resolveImports({
      root: process.cwd(),
    }),
  )(__dirname, undefined);

  expect(resulting).toMatchSnapshot();
});
