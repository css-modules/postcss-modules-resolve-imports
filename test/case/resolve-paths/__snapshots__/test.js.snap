// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resolve-paths 1`] = `
"/*  imported from lib/nested.css  */

/* should be './lib/reset1.css' */
@import url('./lib/reset1.css');
/* should be './lib/reset2.css' */
@import './lib/reset2.css';
/* should be 'lib/anything-non-root1' */
@import url('lib/anything-non-root1');
/* should be 'lib/anything-non-root2' */
@import 'lib/anything-non-root2';
/* should be '/root-based1' */
@import url('/root-based1');
/* should be '/root-based2' */
@import '/root-based2';

._nested_url-decl
{
  /* should be ./lib/spin-icon1.svg */
  background: url(./lib/spin-icon1.svg) 0 0 no-repeat;
  /* should be ./lib/spin-icon2.svg */
  background: url('./lib/spin-icon2.svg') 0 0 no-repeat;
}

/*  imported from source.css  */

._source_relative
{
}
"
`;

exports[`resolve-paths custom root 1`] = `
"/*  imported from test/case/resolve-paths/lib/nested.css  */

/* should be './lib/reset1.css' */
@import url('./test/case/resolve-paths/lib/reset1.css');
/* should be './lib/reset2.css' */
@import './test/case/resolve-paths/lib/reset2.css';
/* should be 'lib/anything-non-root1' */
@import url('test/case/resolve-paths/lib/anything-non-root1');
/* should be 'lib/anything-non-root2' */
@import 'test/case/resolve-paths/lib/anything-non-root2';
/* should be '/root-based1' */
@import url('/root-based1');
/* should be '/root-based2' */
@import '/root-based2';

._nested_url-decl
{
  /* should be ./lib/spin-icon1.svg */
  background: url(./test/case/resolve-paths/lib/spin-icon1.svg) 0 0 no-repeat;
  /* should be ./lib/spin-icon2.svg */
  background: url('./test/case/resolve-paths/lib/spin-icon2.svg') 0 0 no-repeat;
}

/*  imported from test/case/resolve-paths/source.css  */

._source_relative
{
}
"
`;
